webui/react/src/services/decoder.ts

```diff
@@ -117,37 +117,81 @@ export const jsonToGenericCommand = (data: unknown, type: types.CommandType): ty
   return command as types.Command;
 };
 
-const jsonToGenericCommands = (data: unknown, type: types.CommandType): types.Command[] => {
-  const io = ioTypes.decode<ioTypes.ioTypeGenericCommands>(ioTypes.ioGenericCommands, data);
-  return Object.keys(io).map(genericCommandId => {
-    return jsonToGenericCommand(io[genericCommandId], type);
-  });
-};
+const mapV1ContainerStateToCommandState =
```

Comment:
should this use `ResourceState` type?

Reply:
CommandTask state is defined as a CommandState. Should I change that?

Reply:
you're right. CommandState works well here. https://github.com/determined-ai/determined/blob/master/master/internal/command/summary.go

ResourceState is container state + 'Pending" which is a special case that only happens when launching new commands. I think we should do below is that if command.container is missing set the state to `CommandState.PENDING` . would this cover number 1 here https://github.com/determined-ai/determined/pull/1900#issuecomment-771903472 ?

Reply:
Yes, pushing a fix soon.